### ISSUE TITLE
keys not refreshing at the right time

### ISSUE THREAD

**Issue by Gujal00** [Ciarands/vidsrc-keys#1]
The keys dont seem to update at 1 minute past the hour as intended. Can you check please

at 00:52 UTC
in vidsrc-keys `["6BaQS8MmTwl8X9Nz","JPzQzefFPBJLSDYI"]`
in worstsourcekeys `["kL9N2kqrwcwi6OBw","tCxQndnrA1c8WvOp"]`

**Comment by Gujal00** [comment]
the last key update was at 01:23 UTC, as the keys changed at 01:01, the keys from repo were not working until they got updated at at 01:23. Please update the keys closer to the hour

update:
The last update was at 04:19 UTC and has now no updates for over 5 hours.

**Comment by Ciarands** [comment]
Hey @Gujal00, very sorry about this! 
I didnt manage to migrate my extractor onto a VPS as originally planned, as my method couldnt be ran on ARM (which is all I had available at the time), so I ended up just paying for workflow runs just for yesterday.

As you might know workflows are painfully bad when it comes to scheduling, I had the cron set to run every 5 minutes, to which it would only run every 15-30 minutes, this is pathetically bad for a paid service, its a joke that its a worse experience as a paying customer than it is for their free tier in a public repo, this is why it was so badly out of sync at `01:23 UTC`!

Anyway, since I only planned to run this overnight before finding a better solution, I quckly calculated my overhead costs and set that as my billing limit.
When I woke up to my surprise it had only been running for half the night before hitting my set limit (after also charging me well over this limit), this is because Github in their infinite wisdom decided rather than bill you for your actual useage time they should instead round up to the next minute after each run and charge you for a full minute each time! I was being charged double what I had calculated.

I have also just fixed an edgecase where the key function identifier would be in a callback, it should (hopefully) be fully functional now.
I would like to sincerely appologise for any inconvenience caused.
Thanks for your patience!

*TL;DR - Github workflows are laughably awful, you should never use them in a private repo.*

**Comment by Gujal00** [comment]
@Ciarands thanks for running this, I apologise if I came across as whiny. You are doing a stellar job and we are grateful for it.

As mentioned earlier by https://github.com/Claudemirovsky/worstsource-keys/issues/2 you do not have to run the job every 5 minutes. The keys change once every hour 10 seconds after the hour. So if you set the cron to run once an hour at one minute past the hour, it will suffice.

And yes Github workflows are expensive and very lazy when used in private repos, claudemirovsky made several changes to the actions before he settled on having the action run, then wait for 60 minutes, trigger the next action and close. He can do that as it is a public repo, no idea if it would be economical on a private repo

**Comment by Ciarands** [comment]
@Gujal00 don't worry you didn't come across like that at all.
I had it running every 5 minutes to try and minimize how long the keys were out of sync. It isn't really viable to use a workaround like the one in Claude's repo, as you're being charged the entire time it's sleeping.
Anyway, it's a non-issue now. I have the key extractor setup in a docker container, handled by a service on an oracle always free tier server.
